pkg/cloudprovider/providers/aws/aws_routes.go

```diff
@@ -56,17 +62,33 @@ func (s *AWSCloud) ListRoutes(clusterName string) ([]*cloudprovider.Route, error
 	}
 
 	var routes []*cloudprovider.Route
+	var instanceIDs []*string
+
 	for _, r := range table.Routes {
-		instanceID := orEmpty(r.InstanceId)
-		destinationCIDR := orEmpty(r.DestinationCidrBlock)
+		instanceID, destinationCIDR := getInstanceAndCIDR(r)
 
 		if instanceID == "" || destinationCIDR == "" {
 			continue
 		}
 
-		instance, err := s.getInstanceById(instanceID)
-		if err != nil {
-			return nil, err
+		instanceIDs = append(instanceIDs, &instanceID)
+	}
+
+	instances, err := s.getInstancesByIDs(instanceIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range table.Routes {
+		instanceID, destinationCIDR := getInstanceAndCIDR(r)
+
+		if instanceID == "" || destinationCIDR == "" {
+			continue
+		}
+
+		instance, found := instances[instanceID]
+		if !found {
+			fmt.Errorf("unable to find instance ID %s in the list of instances being routed to", instanceID)
```

Comment:
You need to stop processing here, because instance would be nil (and that would cause a panic when we checked PrivateDnsName).  You just need a 'continue' line here.